FilmFestivalLoader/Shared/parse_tools.py

```diff
@@ -52,12 +52,13 @@ def try_parse_festival_sites(parser, festival_data, error_collector, debug_recor
 
 
 class FileKeeper:
-    base_dir = os.path.expanduser(f'~/{config()["Paths"]["FestivalRootDirectory"]}')
+    basedir = os.path.expanduser(f'~/{config()["Paths"]["FestivalRootDirectory"]}')
     common_data_dir = os.path.expanduser(f'~/{config()["Paths"]["CommonDataDirectory"]}')
 
-    def __init__(self, festival, year):
+    def __init__(self, festival, year, basedir=None):
         # Define directories.
-        self.festival_dir = os.path.join(self.base_dir, f'{festival}')
+        self.basedir = basedir or self.basedir
```

Comment:
Please consider to add a comment that explains that the `basedir` option should only be used for testing.

Reply:
Done.